### PR TITLE
Get participants for challenge

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ParticipationRepository.java [hunk 1]
```diff
@@ -18,4 +19,12 @@
...
             "AND   participation.key.challenge.id = :challenge")
     Participation findOneByUserAndChallenge(@Param("user") UUID user, @Param("challenge") UUID challenge);
+
+    @Query("SELECT participation FROM Participation      participation " +
+            "LEFT JOIN FETCH participation.key.challenge challenge " +
+            "LEFT JOIN FETCH participation.key.user      user " +
+            "LEFT JOIN FETCH participation.team          team " +
+            "LEFT JOIN FETCH team.members                teamMember " +
+            "WHERE  participation.key.challenge.canonicalName = :challenge")
+    List<Participation> findAllByChallengeCanonicalName(@Param("challenge") String challenge);
```
Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ParticipationRepository.java]: No order implied by the query but returning a list. Why?

Comment (victorbalan) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ParticipationRepository.java]: what should i order by?

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/dto/participation/ParticipationShowDto.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import uno.cod.platform.server.core.domain.Participation;
+import uno.cod.platform.server.core.dto.challenge.ChallengeShortShowDto;
+import uno.cod.platform.server.core.dto.team.TeamShowDto;
+import uno.cod.platform.server.core.dto.user.UserShowDto;
+
+import java.util.Date;
+
+public class ParticipationShowDto {
+    private UserShowDto user;
+    private TeamShowDto team;
```
Comment (Exordian) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/dto/participation/ParticipationShowDto.java]: we shound't embedd everything, we should return either just the things we need (names?) or just return ID's and fetch the details from the designated endpoint and store those things in our localStorage or indexdb, teams and users should be stored in indexdb imo

Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/dto/participation/ParticipationShowDto.java]: So you are suggesting to defer a change that is straightforward but might add a DTO that is a bit too large until we implemented some IndexDB magic on the client side? WTF?

**Comment by victorbalan** [comment]
will pr to dev

**Pull request closed without merging**